src/rest-server/src/config/yarn.js

```diff
@@ -45,4 +58,18 @@ if (error) {
 }
 yarnConfig = value;
 
+
+// framework launcher health check
+if (config.env !== 'test') {
+  unirest.get(yarnConfig.yarnVcInfoPath)
+  .timeout(2000)
+  .end((res) => {
+    if (res.status === 200) {
+      logger.info('connected to yarn successfully');
+    } else {
+      throw new Error('cannot connect to yarn');
+    }
+  });
+}
+
```

Comment:
@fanyangCS 

Shall we add a yarn dependency to rest server?

Reply:
good question. the desired behavior is: if yarn is available, it works. if yarn is unavailable, the system says it doesn't support the feature yet.

---
In reply to: [239478932](https://github.com/Microsoft/pai/pull/1831#discussion_r239478932) [](ancestors = 239478932)

Reply:
@Gerhut @fanyangCS 
I will move the validation to somewhere, but I still have a question.
Do we consider `vc` as the same concept of `hadoop queue`, or a abstract interface, `hadoop queue` is one implementation? Now I think answer is the former, so moving yarn check under `vc` related API sounds reasonable?